R/integrate_to_ppi.R

```diff
@@ -157,18 +157,54 @@ integrate_to_ppi <- function(pvol, vp, nx = 100, ny = 100, xlim, ylim, zlim = c(
   if (!is.pvol(pvol)) stop("'pvol' should be an object of class pvol")
   if (!is.vp(vp)) stop("'vp' should be an object of class vp")
   if (!assertthat::is.number(nx) && missing(res)) stop("'nx' should be an integer")
-  if (!assertthat::is.number(ny) && missing(res)) stop("ny' should be an integer")
+  if (!assertthat::is.number(ny) && missing(res)) stop("'ny' should be an integer")
   if (!missing(xlim)) {
-    if (length(xlim) != 2 & !is.numeric(xlim)) stop("'xlim' should be an integer vector of length two")
-    if (is.na(xlim[1]) | is.na(xlim[2]) | xlim[1] > xlim[2]) stop("'xlim' should be a vector with two numeric values for upper and lower bound")
+    assertthat::assert_that(
+      is.numeric(xlim),
+      msg = "'xlim' should be an integer vector of length two")
+    assertthat::assert_that(
+      length(xlim) == 2,
+      msg = "'xlim' should be an integer vector of length two")
+    assertthat::assert_that(
+      all(!is.na(xlim)),
+      msg = "'xlim' should be a vector with two numeric values for upper and lower bound")
+    assertthat::assert_that(
+      xlim[1] < xlim [2],
```

Comment:
These all seemed flipped compared to the examples, so I switched them over from `xlim[1] > xlim[2]` to `xlim[1] < xlim[2]`

Please review carefully. The same for `xlim` and `zlim`. What should happen if they are the same (range is 0)?